src/Workspaces/Core/Portable/Workspace/Solution/VersionStamp.cs

```diff
@@ -44,15 +44,16 @@ private VersionStamp(DateTime utcLastModified)
         }
 
         private VersionStamp(DateTime utcLastModified, int localIncrement)
+            : this(utcLastModified, localIncrement, GetNextGlobalVersion())
         {
-            _utcLastModified = utcLastModified;
-            _localIncrement = localIncrement;
-            _globalIncrement = GetNextGlobalVersion();
         }
 
         private VersionStamp(DateTime utcLastModified, int localIncrement, int globalIncrement)
         {
-            Contract.ThrowIfFalse(utcLastModified == default || utcLastModified.Kind == DateTimeKind.Utc);
+            if (utcLastModified != default && utcLastModified.Kind != DateTimeKind.Utc)
```

Comment:
💡 This change seems unnecessary and cascades to a bunch of additional unnecessary string resources. Recommend reverting to use `Contract.ThrowIfFalse` since this condition is never expected.

Reply:
VersionStamp is public Api, and when invalid argument is given, we throw with argument exception.